crates/pallets/starknet/src/lib.rs

```diff
@@ -330,13 +330,12 @@ pub mod pallet {
     #[pallet::getter(fn seq_addr_update)]
     pub type SeqAddrUpdate<T: Config> = StorageValue<_, bool, ValueQuery>;
 
-    /// Mapping from Starknet class hash to contract class.
-    /// Safe to use `Identity` as the key is already a hash.
+    /// Information about processed L1 Messages
+    /// Based on Nonce value.
     #[pallet::storage]
     #[pallet::unbounded]
     #[pallet::getter(fn l1_messages)]
-    pub(super) type L1Messages<T: Config> =
-        StorageMap<_, Blake2_128Concat, Felt252Wrapper, TransactionHash, OptionQuery>;
+    pub(super) type L1Messages<T: Config> = StorageMap<_, Blake2_128Concat, Nonce, (), OptionQuery>;
```

Comment:
https://docs.substrate.io/build/runtime-storage/#hashing-algorithms

Since the caller cannot forge the key values himself (they have to be nonce), we can use https://paritytech.github.io/polkadot-sdk/master/frame_support/struct.Twox64Concat.html instead